### ISSUE TITLE
Cannot display CJK properly on linux.

### ISSUE THREAD

**Issue by taro-yamada** [godotengine/godot#2491]
**OS:** linux(archlinux)
**godot version:** 1.1-stable and 2.0-alpha-r1660.889d21e

Firstly, I'm not a native speaker of English , So I'm sorry if my English is hard to understand. 

I know that the IME on linux is not supported yet.
So I tried to copy and paste japanese word `テスト`(which means `test`) to Label , Button , Script editor , etc.
But both versions of godot display it as `\u30c6\u30b9\u30c8`.

I read some forum and issuie, and imported the font for everyday use.
Then godot uses the specified font for Latin characters.
But, CJK characters are not yet displayed properly.

**Comment by okamstudio** [comment]
Hi Taro! I'm not exactly sure why the input is converted to those
characters, can test I guess. Copy the katakana from your mail to godot
works for me though on Ubuntu (if using a Japanese font)

We never implemented IME simply because we don't know how Asian users
utilize IME natively, so it's really difficult for us to do it.
Version 2.1 (next after current) is planned to have proper IME, but will
definitely need help with this


On Sat, Sep 19, 2015 at 3:34 AM, taro-yamada <notifications@github.com>
wrote:

> *OS:* linux(archlinux)
> *godot version:* 1.1-stable and 2.0-alpha-r1660.889d21e
>
> Firstly, I'm not a native speaker of English , So I'm sorry if my English
> is hard to understand.
>
> I know that the IME on linux is not supported yet.
> So I tried to copy and paste japanese word テスト(which means test) to Label
> , Button , Script editor , etc.
> But both versions of godot display it as \u30c6\u30b9\u30c8.
>
> I read some forum and issuie, and imported the font for everyday use.
> Then godot uses the specified font for Latin characters.
> But, CJK characters are not yet displayed properly.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/issues/2491>.
>



-- 

*OkamStudio*

**Comment by taro-yamada** [comment]
Thank you for your prompt reply.
I tried on Ubuntu 15.04 64bit and godot1.1-stable  but the same behavior occured for me ;-(

Would you tell me which version of ubuntu/godot are you using?

**Comment by akien-mga** [comment]
I can confirm that there are issues not only with CJK characters, but also with the copy-pasting of any non-ascii character. For example I can input "ç" in the editor without issues, and it's displayed properly. But if I try to copy-paste it from Firefox into the editor then there is no output. And with CJK characters I get the same \u30c6, etc. as above. When copy-pasting from other applications, I guess different behaviours: copy-paste from Libreoffice (GTK, like Firefox) does not work for "ç", and adds a "?" for CJK characters; copy-paste from Kate (Qt) works for "ç" (Latin extended), but no output for CJK...

I tried with various fonts (Free Mono, Google Droid Sans). I tried Google Droid Japanse but got an editor crash when selecting "Unicode" in the font importer.
